tests/library/export-test.slang

```diff
@@ -0,0 +1,27 @@
+// export-test.slang
+
+// A test to try out the basics of using libraries that can be linked (by downstream tools), via the HLSL `export` keyword.
+// (This is in contrast to Slang modules and Slangs own linkage mechanisms)
+
+// This didn't work for lib_6_2 when compiling via DXC (!). Even though it's stated elsewhere the feature is available from 6.1
+
+//TEST:COMPILE: tests/library/export-library.slang -profile lib_6_3 -target dxil -o tests/library/export-library.dxil
+//TEST:COMPILE: tests/library/export-test.slang -entry computeMain -profile cs_6_3 -target dxil -r tests/library/export-library.dxil -o tests/library/export-test.dxil 
+
+// Test the produced kernel.
+
+//TEST:COMPARE_COMPUTE_EX:-slang -compute -profile cs_6_3 -dx12 -use-dxil -shaderobj -Xslang... -r tests/library/export-library.dxil -X.
+
+extern int foo(int a);
```

Comment:
Just to confirm: the rules for the HLSL/dxc `export` are that the function needs to be marked as `export` in the translation unit that *defines* it, but doesn't need to be marked `export` in a translation unit that simply declares and then imports it?

Reply:
I believe that is the case.